core/store/src/main/java/uk/gov/gchq/gaffer/store/Store.java

```diff
@@ -401,15 +466,21 @@ protected JobDetail executeJob(final OperationChain<?> operationChain, final Con
                 LOGGER.warn("Operation chain job failed to execute", e);
                 addOrUpdateJobDetail(operationChain, context, e.getMessage(), JobStatus.FAILED);
             }
-        };
+        });
 
-        executorService.execute(runnable);
+        return jobDetail;
+    }
 
-        return initialJobDetail;
+    public void runAsync(final Runnable runnable) throws StoreException {
```

Comment:
Changing this public interface to add a CheckedException isn't backwards compatible.